pandas/core/indexing.py

```diff
@@ -2429,6 +2428,9 @@ def __getitem__(self, key):
         return super().__getitem__(key)
 
     def __setitem__(self, key, value):
+        # raises exception if key does not exist
+        self.__getitem__(key)
```

Comment:
Maybe we could write it as `assert self.__getitem__(key)`. Then the comment would not be needed because the code would speak for itself. Or does that have a drawback?

Reply:
assert checks for truthiness, so this would not be equivalent (e.g. if getitem here returned False).

Reply:
Woh, of course. What was I thinking??